pkg/status/workstatus.go

```diff
@@ -167,3 +171,26 @@ func runtimeObjectToWorkStatusRef(obj runtime.Object) (*workStatusRef, error) {
 		SourceObjectIdentifier: objIdentifier,
 	}, nil
 }
+
+func getObjectStatusLastUpdateTime(metaObj metav1.Object) *metav1.Time {
+	if managedFields := metaObj.GetManagedFields(); managedFields != nil && len(managedFields) > 0 {
```

Comment:
`range` has defined behavior, which is the convenient behavior in this case, when the slice is `nil` or empty.